drivers/bluetooth/hci/bluenrg_uart.c

```diff
@@ -0,0 +1,864 @@
+/*
+ * Copyright (c) 2021 Marek Matej.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT st_bluenrg2_uart
+
+#include <zephyr.h>
+#include <string.h>
+#include <init.h>
+#include <drivers/uart.h>
+#include <drivers/gpio.h>
+#include <sys/byteorder.h>
+#include <sys/ring_buffer.h>
+
+#include <logging/log.h>
+
+#include <devicetree.h>
+
+#include <sys/util.h>
+
+#include <bluetooth/bluetooth.h>
+#include <bluetooth/hci.h>
+#include <drivers/bluetooth/hci_driver.h>
+
+#define BT_DBG_ENABLED IS_ENABLED(CONFIG_BT_DEBUG_HCI_DRIVER)
+#define LOG_MODULE_NAME bluenrg2
+#include "common/log.h"
+
+/* uart support features */
+#include "../util.h"
+
+#define HCI_NONE		0x00
+#define HCI_CMD			0x01
+#define HCI_ACL			0x02
+#define HCI_SCO			0x03
+#define HCI_EVT			0x04
+#define HCI_ISO			0x05
+#define HCI_VENDOR		0xff
+
+#define RELIABLE_PACKET(_t)  (_t == HCI_CMD || \
+								_t == HCI_ACL || \
+								_t = HCI_EVT || \
+								_t == HCI_ISO || \
+								_t == HCI_VENDOR)
+
+#if defined(CONFIG_BT_BLUENRG_ACI)
+
+/* From ST bluenrg1_gatt_aci.c */
+#define BLUENRG_ACI_GATT_INIT						BT_OP(BT_OGF_VS, 0x101)
+#define BLUENRG_ACI_GATT_ADD_SERVICE				BT_OP(BT_OGF_VS, 0x102)
+#define BLUENRG_ACI_GATT_INCLUDE_SERVICE			BT_OP(BT_OGF_VS, 0x103)
+#define BLUENRG_ACI_GATT_ADD_CHAR					BT_OP(BT_OGF_VS, 0x104)
+#define BLUENRG_ACI_GATT_ADD_CHAR_DESC				BT_OP(BT_OGF_VS, 0x105)
+#define BLUENRG_ACI_GATT_UPDATE_CHAR_VALUE			BT_OP(BT_OGF_VS, 0x106)
+#define BLUENRG_ACI_GATT_DEL_CHAR					BT_OP(BT_OGF_VS, 0x107)
+#define BLUENRG_ACI_GATT_DEL_SERVICE				BT_OP(BT_OGF_VS, 0x108)
```

Comment:
Are these definitions required ?

Reply:
Only few of them are needed, but I was trying to avoid to make separate header file I put all of them in the source. These ACI defines are taken from the original ST support - some of them are needed for the radio calibration purposes, but most of them comes from the ST example where internal stack of the chip is used. 
I should probably reduce the number of defines to bare minimum.